tests/_async/test_document.py

```diff
@@ -15,12 +15,18 @@
 #  specific language governing permissions and limitations
 #  under the License.
 
+# this file creates several documents using bad or no types because
+# these are still supported and should be kept functional in spite
+# of not having appropriate type hints. For that reason the comment
+# below disables many mypy checks that fails as a result of this.
+# mypy: disable-error-code="assignment, index, arg-type, call-arg, operator, comparison-overlap, attr-defined"
```

Comment:
This file has a lot of errors because of documents that are defined using old syntax withtout the type hints. I felt it was best to silence those errors and keep some of those older syntax examples.